controllers/alertsconfig_controller.go

```diff
@@ -132,15 +131,15 @@ func (r *AlertsConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request
 			// 2. Wrong alert name and user is going to correct
 			// Ideal way to handle this is to make the alert config status to error and requeue it once in 5 mins or so instead of standard kube builder requeue time
 			// Update the status and retry it
-			return r.PatchIndividualAlertsConfigError(ctx, &alertsConfig, alertName, alertmanagerv1alpha1.Error, err)
+			return r.PatchIndividualAlertsConfigError(ctx, &alertsConfig, alertName, alertmanagerv1alpha1.Error, err, reqChecksum)
```

Comment:
if you change checksum for error state too- what is happening in retry case? Does the error retry actually going through?

Reply:
You are right, the retry logic will skip since the checkSum is equal, how about we add one more condition like we did for wavefrontalerts [here](https://github.com/keikoproj/alert-manager/blob/72b7cf458275d2f3bb853c989f9c53709fadcc4e/controllers/wavefrontalert_controller.go#L131) 
```
		if exist && alertHashMap[alertName].LastChangeChecksum == reqChecksum &&  alertHashMap[alertName].State != alertmanagerv1alpha1.Error {
			log.V(1).Info("checksum is equal so there is no change. skipping", "alertName", alertName)
			//skip it
			continue
		}
```

We always proceed when the state is error, but we need to update lastChecksum to be able to proceed the corrected spec, let me know your thoughts

Reply:
It is either or correct? if you change if condition to look for Error state that should be sufficient to to proceed further

Reply:
Sorry what did you mean by ```It is either or correct?```

Reply:
I updated the code and test it again:
1. When the alertsconfig in error state, the controller will retry
2. When we update the alerts to fix the error, the flow went through and corrected the state to be Ready

Reply:
I meant you don't need to update the requestchecksum for error case. Just change the if condition something like below

```
if alertHashMap[alertName].State != alertmanagerv1alpha1.Error  {
		if exist && alertHashMap[alertName].LastChangeChecksum == reqChecksum {
			log.V(1).Info("checksum is equal so there is no change. skipping", "alertName", alertName)
			//skip it
			continue
		}
}
```

Reply:
Updated the code and tested again, it worked as expected including retry on error, please take a look